lib/Sema/CSSimplify.cpp

```diff
@@ -960,9 +960,26 @@ ConstraintSystem::TypeMatchResult constraints::matchCallArguments(
 
   for (unsigned paramIdx = 0, numParams = parameterBindings.size();
        paramIdx != numParams; ++paramIdx){
-    // Skip unfulfilled parameters. There's nothing to do for them.
-    if (parameterBindings[paramIdx].empty())
+    // If a parameter is unfulfilled, validate the default argument
+    if (parameterBindings[paramIdx].empty()) {
+      if (!isa<FuncDecl>(callee))
+        continue;
+
+      auto allParams = dyn_cast<FuncDecl>(callee)->getParameters()->getArray();
+      for (auto param : allParams) {
+        if (param->isDefaultArgument()) {
+          auto defaultValueExpr = param->getDefaultValue();
+          cs.generateConstraints(defaultValueExpr, param->getDeclContext());
+
+          auto defaultTy = cs.getType(defaultValueExpr);
+          auto paramTy = param->getType();
+          cs.addConstraint(ConstraintKind::ArgumentConversion, defaultTy,
+                           paramTy, locator);
```

Comment:
To make this even better you can add a new path element `default argument` so this failure could be easily "fixed" and diagnosed by `repairFailures`.